src/dateparser/dateparser.js

```diff
@@ -53,12 +53,16 @@ angular.module('ui.bootstrap.dateparser', [])
       regex: '(?:0|1)[0-9]|2[0-3]',
       apply: function(value) { this.hours = +value; }
     },
+    'hh': {
+      regex: '0[0-9]{1}|1[0-2]{1}',
+      apply: function(value) { this.hours = +value; }
+    },
     'H': {
       regex: '1?[0-9]|2[0-3]',
       apply: function(value) { this.hours = +value; }
     },
     'mm': {
-      regex: '[0-5][0-9]',
+      regex: '[0-5]{1}[0-9]{1}',
```

Comment:
Revert this change - this is no different than the previous version, as tested [here](http://regexpal.com/).